flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -1134,4 +1141,28 @@ private Configuration getOperatorRestConfig(Configuration origConfig) throws IOE
                         });
         return conf;
     }
+
+    protected void scaleJmToZero(
+            EditReplacePatchable<Deployment> jmDeployment, String namespace, String clusterId) {
+        LOG.info("Scaling down JM deployment to 0 before deletion");
+        try {
+
+            // We use patching instead of scaling to avoid the need for new permissions
+            var patch = new DeploymentBuilder().editOrNewSpec().withReplicas(0).endSpec().build();
+            jmDeployment.patch(PatchContext.of(PatchType.JSON_MERGE), patch);
+            kubernetesClient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withLabels(KubernetesUtils.getJobManagerSelectors(clusterId))
```

Comment:
`getJobManagerSelectors()` will not work for standalone clusters, as the `type` label will be different. This should be `StandaloneKubernetesUtils#getJobManagerSelectors` in that case.